packages/container/lib/container.js

```diff
@@ -127,9 +135,86 @@ Container.prototype = {
    */
   lookupFactory(fullName, options) {
     assert('fullName must be a proper full name', this.registry.validateFullName(fullName));
+
+    if (isFeatureEnabled('container-factoryFor')) {
+      deprecate('Using "_lookupFactory" is deprecated. Please use container.factoryFor instead.', false, { id: 'container-lookupFactory', until: '2.12.0', url: 'TODO' });
+    }
+
     return factoryFor(this, this.registry.normalize(fullName), options);
   },
 
+  factoryFor(fullName, options = {}) {
+    let container = this;
+    let normalizedName = this.registry.normalize(fullName);
+    assert('fullName must be a proper full name', this.registry.validateFullName(normalizedName));
+
+    if (options.source) {
+      normalizedName = this.registry.expandLocalLookup(fullName, options);
+      // if expandLocalLookup returns falsey, we do not support local lookup
+      if (!normalizedName) { return; }
+    }
+
+    let factory = this.registry.resolve(normalizedName);
+
+    if (factory === undefined) { return; }
+
+    let manager = {
+      class: factory,
+      create(options = {}) {
```

Comment:
Not sure if I follow. Can we not just transition all the internal stuff over to `factoryFor`?